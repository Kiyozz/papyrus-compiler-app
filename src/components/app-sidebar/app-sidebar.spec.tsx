import { shallow } from 'enzyme'
import React from 'react'
import AppSidebarLink from '../app-sidebar-link/app-sidebar-link'
import AppSidebar from './app-sidebar'

describe('<AppSidebar>', () => {
  it('renders correctly', () => {
    const component = (
      <AppSidebar />
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = shallow((
      <AppSidebar />
    ))

    expect(component).toMatchSnapshot()
    expect(component.find(AppSidebarLink)).toHaveLength(3)
  })
})
