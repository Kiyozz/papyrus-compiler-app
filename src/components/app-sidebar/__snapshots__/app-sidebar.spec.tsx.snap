// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSidebar> matches snapshot 1`] = `
<div
  className="app-sidebar"
>
  <div
    className="app-sidebar-logo"
  >
    <img
      alt="app logo"
      src="papyrus-compiler-app-@2x.png"
    />
  </div>
  <div
    className="app-sidebar-links"
  >
    <AppSidebarLink
      to="/compilation"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="file-code"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </AppSidebarLink>
    <AppSidebarLink
      to="/groups"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="layer-group"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </AppSidebarLink>
    <AppSidebarLink
      to="/settings"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="cog"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </AppSidebarLink>
  </div>
</div>
`;
